compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/JvmSymbols.kt

```diff
@@ -323,4 +323,57 @@ class JvmSymbols(
                 returnType = javaLangClass.typeWith()
             }
         }.symbol
+
+    val spreadBuilder = createClass(FqName("kotlin.jvm.internal.SpreadBuilder")) { klass ->
+        klass.addConstructor().apply {
+            addValueParameter("size", irBuiltIns.intType)
+        }
+
+        klass.addFunction("addSpread", irBuiltIns.unitType).apply {
+            addValueParameter("container", irBuiltIns.anyNType)
+        }
+
+        klass.addFunction("add", irBuiltIns.unitType).apply {
+            addValueParameter("element", irBuiltIns.anyNType)
+        }
+
+        klass.addFunction("size", irBuiltIns.intType)
+
+        klass.addFunction("toArray", irBuiltIns.arrayClass.typeWith(irBuiltIns.anyNType)).apply {
+            addValueParameter("a", irBuiltIns.arrayClass.typeWith(irBuiltIns.anyNType))
```

Comment:
Would it be simpler to add `toArray(): Array<T>` in `SpreadBuilder`, instead of using `toArray(array<T>): Array<T>`?

Reply:
It would be, but the jvm doesn't allow us to have nice things. Briefly, arrays have limited subtyping - you cannot cast `[Ljava/lang/Object;` to, e.g., `[Ljava/lang/Integer;` unless the array was originally allocated with this type. And since we compile generics through erasure, a function like `toArray<T>(): Array<T>` doesn't have enough information to allocate an array of the correct type.